pkg/reconciler/kubernetes/tektonpipeline/tektonpipeline.go

```diff
@@ -411,3 +417,26 @@ func (m *Recorder) logMetrics(status, version string, logger *zap.SugaredLogger)
 		logger.Warnf("Failed to log the metrics : %v", err)
 	}
 }
+
+func (r *Reconciler) markUpgrade(ctx context.Context, tp *v1alpha1.TektonPipeline) error {
+	labels := tp.GetLabels()
+	ver, ok := labels[tektoninstallerset.ReleaseVersionKey]
+	if ok && ver == r.operatorVersion {
+		return nil
+	}
+	if ok && ver != r.operatorVersion {
+		tp.Status.MarkInstallerSetNotReady("Upgrade Pending")
+		tp.Status.MarkNotReady("Upgrade Pending")
+	}
+	if labels == nil {
+		labels = map[string]string{}
+	}
+	labels[tektoninstallerset.ReleaseVersionKey] = r.operatorVersion
+	tp.SetLabels(labels)
+
+	if _, err := r.operatorClientSet.OperatorV1alpha1().TektonPipelines().Update(ctx,
```

Comment:
in future, we should try to convey this through status 
may be `status.phase` and try to minimize updating the "object" in reconciler
..
many times it reconciles bcz of conflict error while updating
...

...

Reply:
agreed :+1: 

this time i was just avoiding adding a new item to all statuses (operatorVersion).